### PR TITLE
Add workflow to install Python code, run tests, and submit coverage report

### PR DESCRIPTION
**Pull request by dylan-myome**
Branching off of a conversation @bwinnacott and I had about adding code coverage as an action, it seemed like it might be more valuable to create a standard workflow for testing and reporting coverage of our Python code repos

### CODE REVIEW

**Review by ScottCruzen0 — APPROVED** [review]
Looks good, probably needs to be merged to try it?

**Review by alavertu — COMMENTED** [review]
It'll be a nice addition to our python CI/CD workflow. I made some comments in-line

**Review by sorozco07 — APPROVED** [review]
This looks good! Although, if we want this to become standard in all repos, then probably also worth adding it to the [myome/example](https://github.com/myome/example/tree/main/.github/workflows) template.

File: .github/workflows/test-and-report.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      run: |
+        black --check ${{ SRC_PATH }}
+
+    - name: Security check with bandit
+      run: |
+        bandit -ll -r ${{ SRC_PATH }}
+
+    - name: Test with pytest
+      run: |
+        make test
```
[the remark this answers is not in the document]
Comment (sorozco07) [reply .github/workflows/test-and-report.yaml:62]: The myome/example repo does have pytest in Makefile. But I agree in removing a Makefile dependency.

[the remark this answers is not in the document]
Comment (reece) [reply .github/workflows/test-and-report.yaml:62]: Let's please converge on the Makefile as the definitive way to run commands. The rationale is that we want people to be able to run these commands locally exactly the same way that they do in github actions. If we put these in the Makefile and actions, then eventually we'll modify one but not the other and get different behavior.

File: .github/workflows/test-and-report.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+      run: |
+        isort --check ${{ SRC_PATH }}
+
+    - name: Format check with black
+      run: |
+        black --check ${{ SRC_PATH }}
+
+    - name: Security check with bandit
+      run: |
+        bandit -ll -r ${{ SRC_PATH }}
```
Comment (reece) [inline .github/workflows/test-and-report.yaml:58]: Please put this command in the Makefile.

File: .github/workflows/test-and-report.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+
+    - name: Format check with isort
+      run: |
+        isort --check ${{ SRC_PATH }}
+
+    - name: Format check with black
+      run: |
+        black --check ${{ SRC_PATH }}
```
Comment (reece) [inline .github/workflows/test-and-report.yaml:54]: Please put this command in the Makefile.

**Review by reece — CHANGES_REQUESTED** [review]
Two requests.

1) Please explain the relationship of this new file to the existing files with overlapping functionality.

2) Please migrate functionality to the top level Makefile so that we're (nearly) guaranteed to run testing, linting, reformatting, etc the same way in local and github actions environments.

**Comment by bwinnacott** [comment]
@reece This would provide a single location from which all python repos can call a quality analysis and testing workflow for CI/CD. As a result, if we need to make any modifications to the methods, we only need to update in one place as opposed to all the repos that use them. So, the python repos that call this workflow would have the `cqa` and `test` jobs removed and combined here instead. Might look something like this: 
```
on:
  push:
  pull_request:

jobs:
  test_and_report:
    uses: myome/actions/.github/workflows/test-and-report.yaml@main
  deploy:
    needs: test_and_report
    ...
```
Combining them into a single job also removes the need to run redundant steps multiple times in a single CI/CD workflow (setup python, install dependencies, etc.). Basically, this would replace and centralize the overlapping functionality in the existing `python-package.yml` files.

**Comment by reece** [comment]
@bwinnacott: Yikes. Sorry for the delay. I didn't see the mention and this didn't come up in my review queue because I'd already reviewed. 

Yep, I get that the goal here is to consolidate actions across repos. I wholly support that.

In addition, I would like to ensure that there's exactly one way to run tests, cqa, reformatting, etc across all packages, and that these commands are identical when executed locally (during development) and in github actions. I've wasted a lot of time debugging failures in one context that aren't in the other because of invocation or config differences.

In practice, I think this means putting the meat of all of these commands in a Makefile and invoking that make target from the action. See my [previous comment](https://github.com/myome/actions/pull/29#pullrequestreview-1266796012).

**Review by alavertu — APPROVED** [review]
lgtm, will be good to consolidate around this and extend later if we'd like

**Pull request merged**